pkg/virt-controller/watch/dra/dra.go

```diff
@@ -0,0 +1,607 @@
+		// The controller we found with this Name is not the same one that the
+		// ControllerRef points to.
+		return nil
+	}
+	return vmi.(*virtv1.VirtualMachineInstance)
+}
+
+func (c *DRAStatusController) Run(threadiness int, stopCh <-chan struct{}) {
+	defer controller.HandlePanic()
+	defer c.Queue.ShutDown()
+	log.Log.Info("Starting DRA Status controller")
+
+	// Wait for cache sync before we start the pod controller
+	cache.WaitForCacheSync(stopCh, c.hasSynced)
+
+	// Start the actual work
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+	log.Log.Info("Stopping DRA Status controller")
+}
+
+func (c *DRAStatusController) runWorker() {
+	for c.Execute() {
+	}
+}
+
+var draStatusControllerWorkQueueTracer = &traceUtils.Tracer{Threshold: time.Second}
+
+func (c *DRAStatusController) Execute() bool {
+	if !c.clusterConfig.GPUsWithDRAGateEnabled() {
+		return false
+	}
+	key, quit := c.Queue.Get()
+	if quit {
+		return false
+	}
+
+	draStatusControllerWorkQueueTracer.StartTrace(key, "dra-status-controller VMI workqueue", trace.Field{Key: "Workqueue Key", Value: key})
+	defer draStatusControllerWorkQueueTracer.StopTrace(key)
+
+	defer c.Queue.Done(key)
+	err := c.execute(key)
+
+	if err != nil {
+		log.Log.Reason(err).Infof("reenqueuing VirtualMachineInstance %v", key)
+		c.Queue.AddRateLimited(key)
+	} else {
+		log.Log.V(4).Infof("processed VirtualMachineInstance %v", key)
+		c.Queue.Forget(key)
+	}
+	return true
+}
+
+func (c *DRAStatusController) execute(key string) error {
+	obj, exists, err := c.vmiIndexer.GetByKey(key)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		return nil
+	}
+	vmi := obj.(*virtv1.VirtualMachineInstance)
+	if vmi == nil {
+		return fmt.Errorf("nil vmi reference")
+	}
+	logger := log.Log.Object(vmi)
+
+	if vmi.DeletionTimestamp != nil {
+		// object is being deleted, do not process it
+		log.Log.Info("vmi being deleted, dra status controller skipping")
+		return nil
+	}
+	// Only consider pods which belong to this vmi
+	// excluding unfinalized migration targets from this list.
+	pod, err := controller.CurrentVMIPod(vmi, c.podIndexer)
+	if err != nil {
+		logger.Reason(err).Error("Failed to fetch pods for namespace from cache.")
+		return err
+	}
+	if pod == nil {
+		return fmt.Errorf("nil pod reference for vmi")
+	}
+
+	err = c.updateStatus(logger, vmi, pod)
+	if err != nil {
+		logger.Reason(err).Error("error updating status")
+		return err
+	}
+
+	return nil
+}
+
+func (c *DRAStatusController) updateStatus(logger *log.FilteredLogger, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	if !c.isPodResourceClaimStatusFilled(logger, pod) {
+		logger.V(4).Infof("Waiting for pod %s/%s resource claim status to be filled", pod.Namespace, pod.Name)
+		return nil
+	}
+
+	gpuDeviceInfo, err := c.getGPUDevicesFromVMISpec(vmi)
+	if err != nil {
+		return err
+	}
+
+	gpuStatuses, err := c.getGPUStatuses(gpuDeviceInfo, pod)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Add a `StepTrace` to update status, since there are API calls other than GET.  Something similar to https://github.com/kubevirt/kubevirt/blob/d617cbe5890d7039873c8c87d226b3500c59d47f/pkg/virt-controller/watch/pool/pool.go#L1275